sw/host/opentitanlib/src/io/spi.rs

```diff
@@ -33,6 +74,19 @@ pub enum TransferMode {
     Mode3,
 }
 
+impl FromStr for TransferMode {
+    type Err = SpiError;
+    fn from_str(s: &str) -> std::result::Result<TransferMode, Self::Err> {
+        match s {
+            "Mode0" | "mode0" | "0" => Ok(TransferMode::Mode0),
```

Comment:
I do wish there was a way to use https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case in patterns... oh well.